File: internal/core/ports/repositories/billingcontrol.go
/*
 * Copyright 2023-2025 Eric Moss
 * Licensed under FSL-1.1-ALv2 (Functional Source License 1.1, Apache 2.0 Future)
 * Full license: https://github.com/emoss08/Trenova/blob/master/LICENSE.md */

package repositories

import (
	"context"

	"github.com/emoss08/trenova/internal/core/domain/billing"
	"github.com/emoss08/trenova/pkg/types/pulid"
)

type GetBillingControlRequest struct {
	OrgID  pulid.ID
	BuID   pulid.ID
	UserID pulid.ID
}

type BillingControlRepository interface {
	GetByOrgID(ctx context.Context, orgID pulid.ID) (*billing.BillingControl, error)
	Update(ctx context.Context, bc *billing.BillingControl) (*billing.BillingControl, error)
}

File: ui/src/services/billing-control.ts
/**
 * # Copyright 2023-2025 Eric Moss
 * # Licensed under FSL-1.1-ALv2 (Functional Source License 1.1, Apache 2.0 Future)
 * # Full license: https://github.com/emoss08/trenova/blob/main/LICENSE.md
 */

import { http } from "@/lib/http-client";
import type { BillingControlSchema } from "@/lib/schemas/billing-schema";

export class BillingControlAPI {
  async get() {
    const response = await http.get<BillingControlSchema>("/billing-controls/");
    return response.data;
  }

  async update(data: BillingControlSchema) {
    const response = await http.put<BillingControlSchema>(
      "/billing-controls/",
      data,
    );
    return response.data;
  }
}
